proto/substrait/algebra.proto

```diff
@@ -881,7 +882,21 @@ message Expression {
     message IntervalDayToSecond {
       int32 days = 1;
       int32 seconds = 2;
-      int32 microseconds = 3;
+
+      // oneof ensures that a user can only use precision (and subseconds) or microseconds
+      oneof precision_mode {
+        int32 microseconds = 3 [deprecated = true]; // use precision and subseconds below, they cover and replace microseconds.
+        // Sub-second precision, 0 means the value given is in seconds, 3 is milliseconds, 6 microseconds, 9 is nanoseconds.
+        int32 precision = 4;
+      }
+
+      // the number of fractional seconds using  1e(-precision) units. Should be ignored if microseconds is set above.
+      int64 subseconds = 5;
+    }
+
+    message IntervalCompound {
```

Comment:
Essentially this is struct<iyear, iday>.  To specify it in text one would need to use the nested struct syntax.  Would this be better flattened with the fields from both types included at the top level?

Reply:
Are you talking about the JSON representation or something else? I wasn't really worried about the json representation. Felt like this better clearly communicated the nature as opposed to field copying. It's a weak preference though.

Reply:
Both JSON and the Substrait Text format (which was what I was thinking about since it'll require implementation) would be affected.  I prefer struct<iyear, iday> to the compound approach.  I would also classify my preference as weak.

Reply:
I would expect the substrait text format might have the fields be flattened. I don't think it is critical for it to follow the same structure as the protobufs as long as translation from one to the other is supported, right?